test/Microsoft.ML.CodeAnalyzer.Tests/Code/ContractsCheckTest.cs

```diff
@@ -3,49 +3,67 @@
 
 using System;
-using System.Linq;
+using System.Threading.Tasks;
 using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Testing;
 using Microsoft.ML.CodeAnalyzer.Tests.Helpers;
 using Xunit;
+using VerifyCS = Microsoft.ML.CodeAnalyzer.Tests.Helpers.CSharpCodeFixVerifier<
+    Microsoft.ML.InternalCodeAnalyzer.ContractsCheckAnalyzer,
+    Microsoft.ML.InternalCodeAnalyzer.ContractsCheckNameofFixProvider>;
 
 namespace Microsoft.ML.InternalCodeAnalyzer.Tests
 {
-    public sealed class ContractsCheckTest : DiagnosticVerifier<ContractsCheckAnalyzer>
+    public sealed class ContractsCheckTest
     {
         private readonly Lazy<string> Source = TestUtils.LazySource("ContractsCheckResource.cs");
         private readonly Lazy<string> SourceContracts = TestUtils.LazySource("Contracts.cs");
         private readonly Lazy<string> SourceFriend = TestUtils.LazySource("BestFriendAttribute.cs");
 
         [Fact]
-        public void ContractsCheck()
+        public async Task ContractsCheck()
         {
-            var analyzer = GetCSharpDiagnosticAnalyzer();
-            var diagName = analyzer.SupportedDiagnostics[0];
-            var diagExceptExp = analyzer.SupportedDiagnostics[1];
-            var diagMsg = analyzer.SupportedDiagnostics[2];
-            var diagDecode = analyzer.SupportedDiagnostics[3];
-
             const int basis = 10;
             var expected = new DiagnosticResult[] {
-                diagName.CreateDiagnosticResult(basis + 8, 46, "CheckParam", "paramName", "\"p\""),
-                diagName.CreateDiagnosticResult(basis + 9, 46, "CheckParam", "paramName", "nameof(p) + nameof(p)"),
-                diagName.CreateDiagnosticResult(basis + 11, 28, "CheckValue", "paramName", "\"p\""),
-                diagName.CreateDiagnosticResult(basis + 13, 39, "CheckUserArg", "name", "\"foo\""),
-                diagExceptExp.CreateDiagnosticResult(basis + 15, 13, "Except"),
-                diagExceptExp.CreateDiagnosticResult(basis + 16, 13, "ExceptParam"),
-                diagName.CreateDiagnosticResult(basis + 22, 39, "ExceptParam", "paramName", "\"env\""),
-                diagMsg.CreateDiagnosticResult(basis + 24, 29, "Check", "$\"Hello {foo} is cool\""),
-                diagMsg.CreateDiagnosticResult(basis + 28, 29, "Check", "string.Format(\"Hello {0} is cool\", foo)"),
-                diagMsg.CreateDiagnosticResult(basis + 32, 35, "Check", "\"Less fine: \" + env.GetType().Name"),
-                diagName.CreateDiagnosticResult(basis + 34, 17, "CheckUserArg", "name", "\"p\""),
-                diagDecode.CreateDiagnosticResult(basis + 39, 41, "CheckDecode", "\"This message is suspicious\""),
+                new DiagnosticResult("CS0051", DiagnosticSeverity.Error).WithLocation(14, 16).WithMessage("Inconsistent accessibility: parameter type 'IHostEnvironment' is less accessible than method 'TypeName.TypeName(IHostEnvironment, float, int)'"),
```

Comment:
📝 It was not clear to me if this was a desired error in the test code. It could indicate that the test input has become stale relative to some API and needs to be updated.

Reply:
Very interesting. Absolutely not intended, these may have crept in as the input source files changed. Anyway it should be fixed, though perhaps not part of this PR, since this is about using the proper test harness, and we can hardly expect you to fix problems with our faulty tests. Anyway, I have filed a separate issue #2588 to capture this work. My comment would be the same for all these "compilation error" additions you needed to add, absolutely not the intent of the class. But thank you for finding this, this is definitely something we must do!